StortSpelProjekt/Engine/src/Renderer/QuadManager.cpp

```diff
@@ -0,0 +1,166 @@
+#include "stdafx.h"
+#include <vector>
+#include "QuadManager.h"
+
+#include "../Renderer/Texture/Texture.h"
+#include "../Renderer/DescriptorHeap.h"
+#include "../Renderer/GPUMemory/Resource.h"
+#include "../Renderer/GPUMemory/ShaderResourceView.h"
+#include "../Renderer/Mesh.h"
+#include "../Renderer/Renderer.h"
+
+#include "../Misc/Window.h"
+#include "../Misc/AssetLoader.h"
+
+#include "../Events/EventBus.h"
+
+QuadManager::QuadManager()
+{
+}
+
+QuadManager::~QuadManager()
+{
+	EventBus::GetInstance().Unsubscribe(this, &QuadManager::pressed);
+
+	delete m_pQuad;
+}
+
+void QuadManager::CreateQuad(float2 pos, float2 size, bool clickable, std::wstring texturePath)
+{
+	if (m_pQuad != nullptr)
+	{
+		delete m_pQuad;
+		m_Clickable = false;
+	}
+
+	if (m_pQuadTexture == nullptr && texturePath != L"NONE")
+	{
+		AssetLoader* al = AssetLoader::Get();
+		m_pQuadTexture = al->LoadTexture2D(texturePath);
+	}
```

Comment:
Why is m_pQuad and m_pQuadTexture managed/used in different ways when they should be the same?